api/server/graph.cc

```diff
@@ -608,9 +608,19 @@ std::string Graph::fw_build_rule(const app::Rule &rule) {
 std::string Graph::fw_build_sg(const app::Sg &sg) {
     std::string r;
     for (auto it = sg.rules.begin(); it != sg.rules.end();) {
-        r += "(" + fw_build_rule(it->second) + ")";
+        auto fw_rule = fw_build_rule(it->second);
```

Comment:
do we really need to use auto here ?
I mean, I have no idea what's the type of fw_rule.